python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -471,6 +502,97 @@ def update_masked_image_stats(self, summary, masked_image):
         meanVar, _ = statObj.getResult(afwMath.MEANCLIP)
         summary.meanVar = meanVar
 
+    def update_effective_time_stats(self, summary, exposure):
+        """Compute effective exposure time statistics to estimate depth.
+
+        The effective exposure time is the equivalent shutter open
+        time that would be needed under nominal conditions to give the
+        same signal-to-noise for a point source as what is achieved by
+        the observation of interest. This metric combines measurements
+        of the point-spread function, the sky brightness, and the
+        transparency.
+
+        .. _teff_definitions:
+
+        The effective exposure time and its subcomponents are defined in [1]_
+
+        References
+        ----------
+
+        .. [1] Neilsen, E.H., Bernstein, G., Gruendl, R., and Kent, S. (2016).
+               Limiting Magnitude, \tau, teff, and Image Quality in DES Year 1
+               https://www.osti.gov/biblio/1250877/
+
+
+        Parameters
+        ----------
+        summary : `lsst.afw.image.ExposureSummaryStats`
+            Summary object to update in-place.
+        exposure : `lsst.afw.image.ExposureF`
+            Exposure to grab band and exposure time metadata
+
+        """
+        self.log.info("Updating effective exposure time")
+
+        nan = float("nan")
+        summary.effTime = nan
+        summary.effTimeScalePsfSigma = nan
+        summary.effTimeScaleSkyBg = nan
+        summary.effTimeScaleZeroPoint = nan
+
+        exposureTime = exposure.getInfo().getVisitInfo().getExposureTime()
+        filterLabel = exposure.getFilter()
+        if (filterLabel is None) or (not filterLabel.hasBandLabel):
+            band = None
+        else:
+            band = filterLabel.bandLabel
+
+        # PSF component
+        if np.isnan(summary.psfSigma):
+            self.log.warn("PSF sigma is NaN")
+            f_eff = nan
+        elif band not in self.config.fiducialPsfSigma:
+            self.log.warn(f"Fiducial PSF value not found for {band}")
```

Comment:
Same here.  It _may_ be worth doing a warn above if band is None ... wouldn't that mean that none of this could be computed?  If so, then a warning of "no band associated with this filter, and effective exposure time statistics won't be available" may be okay (and would point to some terrible problem somewhere).

Reply:
Good point. I think that if the band is None then it is probably worth warning and returning with the effTime values set to NaN. On the other hand, if the band exists but is not in (all of) the fiducial dictionaries, it was less clear to me what should be done. I guess the safest thing is to return NaN unless fiducial values exist for *all three* components for the band of interest.